index.d.ts

```diff
@@ -1,146 +1,148 @@
-   */
-  interface EventLoopDelayHistogram {
-    min: number,
-    max: number,
-    mean: number,
-    stddev: number,
-    percentiles: Map<number, number>,
-    exceeds: number,
-  }
-
-  enum GCFlag { // eslint-disable-line no-unused-vars
-    // perf_hooks.constants.NODE_PERFORMANCE_GC_FLAGS_NO
-    no = 'no', // eslint-disable-line no-unused-vars
-    // perf_hooks.constants.NODE_PERFORMANCE_GC_FLAGS_CONSTRUCT_RETAINED
-    constructRetained = 'constructRetained', // eslint-disable-line no-unused-vars
-    // perf_hooks.constants.NODE_PERFORMANCE_GC_FLAGS_FORCED
-    forced = 'forced', // eslint-disable-line no-unused-vars
-    // perf_hooks.constants.NODE_PERFORMANCE_GC_FLAGS_SYNCHRONOUS_PHANTOM_PROCESSING
-    synchronousPhantomProcessing = 'synchronousPhantomProcessing', // eslint-disable-line no-unused-vars
-    // perf_hooks.constants.NODE_PERFORMANCE_GC_FLAGS_ALL_AVAILABLE_GARBAGE
-    allAvailableGarbage = 'allAvailableGarbage', // eslint-disable-line no-unused-vars
-    // perf_hooks.constants.NODE_PERFORMANCE_GC_FLAGS_ALL_EXTERNAL_MEMORY
-    allExternalMemory = 'allExternalMemory', // eslint-disable-line no-unused-vars
-    // perf_hooks.constants.NODE_PERFORMANCE_GC_FLAGS_SCHEDULE_IDLE
-    scheduleIdle = 'scheduleIdle' // eslint-disable-line no-unused-vars
+    * Enable/disable specific metrics
+    */
+  collect?: {
+    cpu?: boolean,
+    eventLoopDelay?: boolean,
+    memory?: boolean,
+    gc?: boolean,
+    activeHandles?: boolean
   }
+}
 
-  enum GCOp { // eslint-disable-line no-unused-vars
-    // perf_hooks.constants.NODE_PERFORMANCE_GC_MAJOR
-    major = 'major', // eslint-disable-line no-unused-vars
-    // perf_hooks.constants.NODE_PERFORMANCE_GC_MINOR
-    minor = 'minor', // eslint-disable-line no-unused-vars
-    // perf_hooks.constants.NODE_PERFORMANCE_GC_INCREMENTAL
-    incremental = 'incremental', // eslint-disable-line no-unused-vars
-    // perf_hooks.constants.NODE_PERFORMANCE_GC_WEAKCB
-    weakCB = 'weakCB' // eslint-disable-line no-unused-vars
-  }
+declare interface CPUMetric {
+  /**
+    * Usage percentage
+    */
+  usage: number,
+  /**
+    * Raw value returned by `process.cpuUsage()`
+    */
+  raw: number
+}
 
-  interface GCOpStats {
-    count: number,
-    total: number,
-    average: number
-  }
+/**
+ * On Node 12 and above this is a Histogram instance from 'perf_hooks'.
+ */
+declare interface EventLoopDelayHistogram {
+  min: number,
+  max: number,
+  mean: number,
+  stddev: number,
+  percentiles: Map<number, number>,
+  exceeds: number,
+}
+declare interface EventLoopDelayMetric {
+  /**
+    * computed delay in milliseconds
+    */
+  computed: number,
+  raw: number | EventLoopDelayHistogram
+}
 
-  interface GCAggregatedEntry extends GCOpStats {
-    flags?: Map<GCFlag, GCOpStats>
-  }
+declare enum GCFlag {
+  // perf_hooks.constants.NODE_PERFORMANCE_GC_FLAGS_NO
+  no = 'no', // eslint-disable-line no-unused-vars
```

Comment:
I know this is part of the `gc_improvements` branch and not this one, but anyway:

enums in TypeScript are recommended to be PascalCase (first letter is a capital) so I would prefer that this be something like:

```ts
declare enum GCFlag {
  // perf_hooks.constants.NODE_PERFORMANCE_GC_FLAGS_NO
  No = 'no',
  // perf_hooks.constants.NODE_PERFORMANCE_GC_FLAGS_CONSTRUCT_RETAINED
  ConstructRetained = 'constructRetained',
  // perf_hooks.constants.NODE_PERFORMANCE_GC_FLAGS_FORCED
  Forced = 'forced',
  // perf_hooks.constants.NODE_PERFORMANCE_GC_FLAGS_SYNCHRONOUS_PHANTOM_PROCESSING
  SynchronousPhantomProcessing = 'synchronousPhantomProcessing',
  // perf_hooks.constants.NODE_PERFORMANCE_GC_FLAGS_ALL_AVAILABLE_GARBAGE
  AllAvailableGarbage = 'allAvailableGarbage',
  // perf_hooks.constants.NODE_PERFORMANCE_GC_FLAGS_ALL_EXTERNAL_MEMORY
  AllExternalMemory = 'allExternalMemory',
  // perf_hooks.constants.NODE_PERFORMANCE_GC_FLAGS_SCHEDULE_IDLE
  ScheduleIdle = 'scheduleIdle'
}
```

Reply:
Also the JSDoc comment style applies here too (see my comment about TypeScript Language Servers further down)